src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AmendReferralService.kt

```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.AmendComplexityLevelDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
+import java.time.OffsetDateTime
+import java.util.*
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class AmendReferralService(
+  val changelogRepository: ChangelogRepository,
+  val referralRepository: ReferralRepository
+) {
+
+  fun updateComplexityLevel(referral: Referral, update: AmendComplexityLevelDTO, actor: AuthUser): AmendComplexityLevelDTO? {
+    val existingDetails = referralRepository.findById(referral.id).get()
+
+    val complexityLevel = existingDetails.complexityLevelIds?.getValue(update.serviceCategoryId)
+
+    existingDetails.complexityLevelIds = mutableMapOf(update.serviceCategoryId to update.complexityLevelId)
+
+    val changelog = Changelog(
+      referral.id,
+      UUID.randomUUID(),
+      "COMPLEXITY_LEVEL",
+      "{\"complexityLevelId\":\"{$complexityLevel}\"}",
```

Comment:
its looking good for the initial cut but we need to have discussion on having generic object which can hold all old values and new values

Reply:
JSON allows top-level lists, so could be `[...]` where lists are `[1,2,3]` and singular values are `[20]`

Or have separate keys to 'hint':
```json
{
  "value": 20,
  "reference": "ddc8ee4a-403e-47cc-9af1-815938429bbc",
  "values": [20, 40],
  "references": ["52a658f6-622b-491e-8085-cea58c9b015b", "4fc163c9-f8fb-4cfd-b107-2cb13583ebf0"]
}
```
with one key required out of the 4

Reply:
Thanks for this - yeah we are now using an array to store the data which we can use the topic in order to work out whether we could have a list or a single value

Reply:
I am proposing something like this

Reply:
I am proposing some thing like this

```json
{
 "complexityLevelIds":[],
"serviceCategoryId":"",
"desiredOutcomeIds":[]
}
```